ygot/struct_validation_map.go

```diff
@@ -475,14 +475,33 @@ func MergeJSON(a, b map[string]interface{}) (map[string]interface{}, error) {
 	return o, nil
 }
 
+// MergeOpt is an interface that is implemented by the options to the MergeStructInto
+// function. It allows user specified options to be propagated to the merge
+// method.
+type MergeOpt interface {
+	// IsMergeOpt is a marker method for each MergeOpt.
+	IsMergeOpt()
+}
+
+// DiffPathOpt is a DiffOpt that allows control of the path behaviour of the
```

Comment:
Update comment from `DiffPathOpt` to cover the fact that this is actually about the `MergeOpt`.